src/Razor/test/Microsoft.VisualStudio.LanguageServices.Razor.Test/Cohost/TestServiceBroker.cs

```diff
@@ -2,19 +2,23 @@
 // Licensed under the MIT license. See License.txt in the project root for license information.
 
 using System;
+using System.IO.Pipelines;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Razor;
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.ExternalAccess.Razor;
 using Microsoft.CodeAnalysis.Remote.Razor;
+using Microsoft.ServiceHub.Framework;
 
 namespace Microsoft.VisualStudio.Razor.LanguageClient.Cohost;
 
-internal sealed class TestServiceBroker : IRazorServiceBroker
+internal sealed class TestServiceBroker : IServiceBroker, IRazorBrokeredServiceInterceptor
```

Comment:
Is it worth separating these two interfaces out to two different implementations? The use of ServiceArgs now seems like it would be easier. Could even use a mock for IServiceBroker since we don't expect it to be called.